ocaml/lambda/matching.mli

```diff
@@ -22,16 +22,27 @@ open Debuginfo.Scoped_location
 (* Right-hand side of a case. *)
 type rhs
 
+(* The free variables in a guarded rhs can be precomputed to be used in an
+   optimization, or uncomputed, in which case the optimization is not applied.
+*)
+type guarded_free_variables =
```

Comment:
I feel that `guarded_free_variables` is difficult to document from the mli. It's really an implementation detail. We could piggy-back off of the fact that we do the optimization for boolean-guarded RHSes and not pattern-guarded ones, and instead have two functions `mk_boolean_guarded_rhs` and `mk_pattern_guarded_rhs`. This would allow us to avoid exposing or documenting this type in the mli.

Reply:
Yeah, agreed that this is easier to explain, modified in this way